scipy/stats/stats.py

```diff
@@ -2896,9 +2896,9 @@ def spearmanr(a, b=None, axis=0):
         `axis`. Both arrays need to have the same length in the `axis`
         dimension.
     axis : int or None, optional
-        If axis=0 (default), then each column represents a variable, with
-        observations in the rows. If axis=0, the relationship is transposed:
-        each row represents a variable, while the columns contain observations.
+        If axis=0 (default), then each row represents a variable, with
+        observations in the columns. If axis=1, the relationship is transposed:
+        each column represents a variable, while the rows contain observations.
         If axis=None, then both arrays will be raveled.
```

Comment:
Are you sure? You may be right, your code does make for some valid arguments, but try stating the axis. When no axis are stated then they are raveled. I am not entirely sure what that means though. I was showing in an issue a few days ago why I believe in correctness of my fix. Try look it up in clised issues.